lib/angular_xss/safe_buffer.rb

```diff
@@ -1,20 +1,36 @@
 ActiveSupport::SafeBuffer.class_eval do
 
-  if private_method_defined? :html_escape_interpolated_argument
+  if private_method_defined?(:html_escape_interpolated_argument) || # Rails < 7
+    private_method_defined?(:explicit_html_escape_interpolated_argument) # Rails >= 7
 
     private
 
-    def html_escape_interpolated_argument_with_angular_xss(arg)
+    def explicit_html_escape_interpolated_argument_with_angular_xss(arg)
       if arg.html_safe?
         arg
       else
-        html_escape_interpolated_argument_without_angular_xss(AngularXss::Escaper.escape(arg))
+        explicit_html_escape_interpolated_argument_without_angular_xss(AngularXss::Escaper.escape(arg))
       end
     end
 
-    alias_method :html_escape_interpolated_argument_without_angular_xss, :html_escape_interpolated_argument
-    alias_method :html_escape_interpolated_argument, :html_escape_interpolated_argument_with_angular_xss
+    if private_method_defined?(:html_escape_interpolated_argument)
+      alias_method :explicit_html_escape_interpolated_argument_without_angular_xss, :html_escape_interpolated_argument
+      alias_method :html_escape_interpolated_argument, :explicit_html_escape_interpolated_argument_with_angular_xss
+    elsif private_method_defined?(:explicit_html_escape_interpolated_argument)
 
-  end
+      def implicit_html_escape_interpolated_argument_with_angular_xss(arg)
```

Comment:
Using a separate `if` statement which checks for the "implicit" version might be clearer:

```ruby
    if private_method_defined?(:html_escape_interpolated_argument)
      alias_method :html_escape...
    elsif private_method_defined?(:explicit_html_escape_interpolated_argument)
      alias_method :explicit_html_escape...
    end

    if private_method_defined?(:implicit_html_escape_interpolated_argument)
      def implicit_html_escape_interpolated_argument_with_angular_xss(arg)
        ...
      end

      alias_method :implicit_html_escape...
    end
```